ansys/mapdl/core/mapdl.py

```diff
@@ -2164,10 +2164,89 @@ def ignore_errors(self):
     def ignore_errors(self, value):
         self._ignore_errors = bool(value)
 
-    def load_table(self, name, array, var1="", var2="", var3=""):
+    def load_array(self, name, array):
+        """Load an array from Python to MAPDL.
+
+        Uses VREAD to transfer the array.
+        The format of the numbers used in the intermediate file is F24.18.
+
+        Parameters
+        ----------
+        name : str
+            An alphanumeric name used to identify this table.  Name
+            may be up to 32 characters, beginning with a letter and
+            containing only letters, numbers, and underscores.
+            Examples: ``"ABC" "A3X" "TOP_END"``.
+
+        array : np.ndarray or List
+            List as a table or ``numpy`` array.
+
+        Examples
+        --------
+        >>> my_conv = np.array([[0, 0.001],
+                                [120, 0.001],
+                                [130, 0.005],
+                                [700, 0.005],
+                                [710, 0.002],
+                                [1000, 0.002]])
+        >>> mapdl.load_array('MY_ARRAY', my_conv)
+        >>> mapdl.parameters['MY_ARRAY']
+        array([[0.0e+00, 1.0e-03],
+                [1.2e+02, 1.0e-03],
+                [1.3e+02, 5.0e-03],
+                [7.0e+02, 5.0e-03],
+                [7.1e+02, 2.0e-03],
+                [1.0e+03, 2.0e-03]])
+        """
+        if not isinstance(array, np.ndarray):
+            raise ValueError("The array should be a Numpy array")
+        if array.ndim != 2:
+            raise NotImplementedError("Only loading of 2D arrays is supported at the moment.")
+
+        imax = array.shape[0]
+        jmax = array.shape[1]
+        if array.ndim == 2:
+            kmax = ""
+        elif array.ndim == 3:
+            kmax = array.shape[2]
+        else:
+            raise ValueError(
+                f"Expecting only a 2D or 3D array, but input contains\n{array.ndim} dimensions"
+            )
+
+        self.dim(name, "ARRAY", imax=array.shape[0], jmax=array.shape[1], kmax="")
+
+        base_name = random_string() + ".txt"
+        filename = os.path.join(tempfile.gettempdir(), base_name)
+        self._log.info(f"Generating file for table in {filename}")
+        logger.info(f"Generating file for table in {filename}")
```

Comment:
Do we need both here?

Reply:
Probably with only one is fine.

Reply:
I removed the ``logger.info`` which is the main one, using the instance logger ``self._log``.